backend/api/Services/KeyVaultService.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Azure.Security.KeyVault.Secrets;
+
+public class KeyVaultService
+{
+    private readonly SecretClient _secretClient;
+
+    public KeyVaultService(SecretClient secretClient)
+    {
+        _secretClient = secretClient;
+    }
+
+    public async Task<string> GetSecret(string secretName)
+    {
+        try
+        {
+            KeyVaultSecret keyVaultSecret = await _secretClient.GetSecretAsync(secretName);
+            return keyVaultSecret.Value;
+        }
+        catch
+        {
+            throw;
+        }
```

Comment:
No need to catch if we just rethrow, `try/catch` can probably be removed

Reply:
I've added a logging statement to show which secret failed to be retrieved. Will give us a good warning if something isn't working.